api/v1alpha1/obtenantoperation_webhook.go

```diff
@@ -228,10 +227,112 @@ func (r *OBTenantOperation) validateMutation() error {
+		}
 	}
-	if len(allErrs) == 0 {
-		return nil
+	if len(allErrs) != 0 {
+		return allErrs.ToAggregate()
+	}
+
+	if obtenant.Status.Status != tenantstatus.Running && !r.Spec.Force {
+		return field.Invalid(field.NewPath("spec").Child("targetTenant"), r.Spec.TargetTenant, "The target tenant is not in running status")
+	}
+
+	switch r.Spec.Type {
+	case constants.TenantOpSetUnitNumber:
+		if r.Spec.UnitNumber == 0 {
+			allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("unitNumber"), "unitNumber is required"))
+		}
+	case constants.TenantOpSetForceDelete:
+		if r.Spec.ForceDelete == nil {
+			allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("forceDelete"), "forceDelete is required"))
+		}
+	case constants.TenantOpSetConnectWhiteList:
+		if r.Spec.ConnectWhiteList == "" {
+			allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("connectWhiteList"), "connectWhiteList is required"))
+		}
+	case constants.TenantOpSetCharset:
+		if r.Spec.Charset == "" {
+			allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("charset"), "charset is required"))
+		}
+	case constants.TenantOpAddResourcePools:
+		if len(r.Spec.AddResourcePools) == 0 {
+			allErrs = append(allErrs, field.Required(field.NewPath("spec").Child("addResourcePools"), "addResourcePools is required"))
+			break
+		}
+		obcluster := &OBCluster{}
+		err := clt.Get(context.Background(), types.NamespacedName{Name: obtenant.Spec.ClusterName, Namespace: r.Namespace}, obcluster)
+		if err != nil {
+			if !apierrors.IsNotFound(err) {
+				allErrs = append(allErrs, field.InternalError(field.NewPath("spec").Child("targetTenant"), err))
+			} else {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("targetTenant"), r.Spec.TargetTenant, "The target tenant does not exist"))
+			}
+		}
+		if obcluster.Spec.Topology == nil {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("targetTenant"), r.Spec.TargetTenant, "The target tenant does not have a topology"))
+		}
+		pools := make(map[string]any)
+		for _, pool := range obtenant.Spec.Pools {
+			pools[pool.Zone] = struct{}{}
+		}
+		for _, pool := range r.Spec.AddResourcePools {
+			if _, ok := pools[pool.Zone]; ok {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("addResourcePools"), r.Spec.AddResourcePools, "The resource pool already exists"))
+			}
+		}
+		poolsInCluster := make(map[string]any, len(obcluster.Spec.Topology))
+		for _, zone := range obcluster.Spec.Topology {
+			poolsInCluster[zone.Zone] = struct{}{}
+		}
+		for _, pool := range r.Spec.AddResourcePools {
+			if _, ok := poolsInCluster[pool.Zone]; !ok {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("spec").Child("addResourcePools"), r.Spec.AddResourcePools, "The resource pool does not exist in the cluster"))
```

Comment:
This should be zones not in obcluster but not pools because you just want to add pools on such zones

and give the exact zone name that is not existed